### DISCUSSION TITLE
Make field presets (date_created, status, ...) customizable

### DISCUSSION THREAD

**Discussion by LuHugo** [directus/directus#19887] [Feature Requests]
### Summary

At this stage, every time you create a data model, you need to manually select Optional System Fields and modify it. So I'm wondering why it's so troublesome. Can we define some preset fields so that the preset fields can be automatically checked when creating a data model.

### Motivation

You can use preset fields when creating a data model, which avoids the need to set a certain field every time you create a data model. For example: I can set each data model to require `createdAt` and `updatedAt` fields. These fields are already A preset is defined, which will be automatically added when creating the data model or cannot be modified (depending on the preset value)

### Detailed Design

Create a preset field. The settable options of this field should be consistent with the functions in the data model. Of course, there are also some additional settings, such as whether it is selected by default when creating the data model, whether it can be reset, etc.

### Must Have:
- A page where you can add preset fields
- A page to manage preset fields
- On the management page, Optional System Fields can be preset and cannot be deleted.
- There is a preset field tab on the create data model page.
- These presets should not affect previously created data models

### Should Have:
- none

### Could Have:
- none

### Won't Have:
- none

### Drawbacks

none

### Alternatives

none

### Adoption Strategy

Because I don't understand the code logic, I can't say exactly whether it is a major event. I guess this shouldn't be a major change in terms of business relevance.

**Comment by DanielBiegler** [comment]
Hi there, I renamed the title to be more descriptive of your suggestion.

This is a cool idea and would probably benefit many people, I like it! We have the default `status` field which uses simple strings like `draft` or `published` and I can imagine it being very useful to customize the preset to use your own status logic with relations for example. Having to duplicate that over from other collections is slightly annoying.

**Reply by LuHugo** [reply]
Yes, now you need to reset some of the same fields every time you create a data model. This feature can indeed eliminate these boring and repetitive tasks, and it should also apply to using the API to create data models.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
